master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -253,11 +254,24 @@ private[celeborn] class Master(
       internalRpcEndpoint)
   }
 
+  private val sendApplicationMetaThreads = conf.masterSendApplicationMetaThreads
+  // Send ApplicationMeta to workers
+  private var sendApplicationMetaExecutor: ExecutorService = _
+  // Maintains the mapping for the workers assigned to each application
+  private val workersAssignedToApp
```

Comment:
Since this map is not stored in ratis, in corner case where leader changes it can happen that two masters send `APPLICATION_META` to the same worker. So I guess we still can't throw Exception in `SecretRegistryImpl#register`?

Also, I don't think storing `workersAssignedToApp` to ratis is a good idea because it can be large.

Reply:
Good point, we should throw an exception in `SecretRegistryImpl` only if the values dont match via equality.

Reply:
Worker is checking whether the application is registered or not before updating the registry. Isn't that sufficient to handle this?

Reply:
I was looking at `SecretRegistryImpl`, where we throw an exception if already present - and can validate if they are different as well.
Looks like worker will have own registry, so that specific aspect will be handled in #2363